packages/tempus/src/Tempus_StepperBDF2ModifierXBase.hpp

```diff
@@ -0,0 +1,133 @@
+// @HEADER
+// ****************************************************************************
+//                Tempus: Copyright (2017) Sandia Corporation
+//
+// Distributed under BSD 3-clause license (See accompanying file Copyright.txt)
+// ****************************************************************************
+// @HEADER
+
+#ifndef Tempus_StepperBDF2ModifierXBase_hpp
+#define Tempus_StepperBDF2ModifierXBase_hpp
+
+#include "Tempus_config.hpp"
+#include "Tempus_SolutionHistory.hpp"
+#include "Tempus_StepperBDF2AppAction.hpp"
+
+
+namespace Tempus {
+
+/** \brief Base ModifierX for StepperBDF2.
+ *
+ *  This class provides a means to modify just the solution values
+ *  (i.e., \f$x\f$ and \f$dot{x}\f$), and nothing else, but time and
+ *  timestep are also provided.
+ *
+ *  Users deriving from this class can access and change the solution
+ *  during the timestep (e.g., limiting the solution for monoticity).
+ *  It is expected that the user knows what changes are allowable without
+ *  affecting the Stepper correctness, performance, accuracy and stability
+ *  (i.e., USER BEWARE!!).
+ *
+ *  Below is the BDF2 algorithm with the locations of the ModifierX calls
+ *  italicized.
+ *
+ *  \f{algorithm}{
+ *  \renewcommand{\thealgorithm}{}
+ *  \caption{BDF2 with the locations of the application actions indicated}
+ *  \begin{algorithmic}[1]
+ *    \State Start with $x_n$, $\Delta t_n$
+ *    \State {\it appAction.execute(solutionHistory, stepper, BEGIN\_STEP)}
+ *    \State Form $f(x_{n},t_{n})$
+ *    \State {\it appAction.execute(solutionHistory, stepper, BEFORE\_EXPLICIT\_EVAL)}
+ *    \State Form $x_n \leftarrow x_{n} + \Delta t_n f(x_{n},t_n)$
+ *    \State {\it appAction.execute(solutionHistory, stepper, END\_STEP)}
+ *  \end{algorithmic}
```

Comment:
Need to make this match the algorithm: BEGIN_STEP, BEFORE_SOLVE, AFTER_SOLVE, and END_STEP.

Reply:
@ccober6 So no get Xdots anywhere?